tfhe-zk-pok/benches/utils.rs

```diff
@@ -0,0 +1,459 @@
+#![allow(non_snake_case)]
```

Comment:
looks to be mostly a copy paste ?

Reply:
I need the types in `tfhe/benches/utilities` to be able to create the json file with the params but I can't use it directly because I am not in the `tfhe` crate. This is mostly a copy paste with only the needed bits and `tfhe` internal types converted to builtin rust types.

I also integrated things from `tfhe-zk-pok/proof/mod.rs` `test` module because benches are not run with `#[cfg(test)]`

Reply:
the utils can be marked with a cfg = "bench" I believe

I think we have something like that in TFHE-rs

as for the utils some work needs to be done with David to make the copied code into a small util crate, we can't copy things when we "just" need to bench something

Reply:
You think about a crate like `tfhe-utils` that would contains `CryptoParametersRecord` and `write_to_json()` function ?

Reply:
potentially, we can't duplicate those everywhere

Reply:
The problem is that the current bench utilities make heavy use of tfhe types so it cannot be used without it, it would need a rework that's out of the scope of this PR I think. Since tfhe itself depend on tfhe-zk-pok it should be written in a way that prevent circular dependencies.

I don't think there is a `cfg(bench)` and the code is already only compiled with benches, the problem is that I can't use code from `cfg(test)` from there.

Reply:
see tfhe/src/core_crypto/commons/ciphertext_modulus.rs:23 fairly sure there is something for benchmarks

Reply:
Are you sure that this works ? I don't find any doc about it, it does not seem to have any effect on my code and cargo warns about it. Maybe there is some special trick in the tfhe crate to make it work ?